### PR TITLE
Update configmap with custom domains

### PR DESCRIPTION
**Pull request by vthapar**
If custom domains are configured update the Coredns and lighthouse dns
configmaps with relevant configuration.

Signed-off-by: Vishal Thapar <5137689+vthapar@users.noreply.github.com>

### CODE REVIEW

File: pkg/controller/servicediscovery/servicediscovery_controller.go [hunk 1]
```diff
@@ -382,31 +374,34 @@ func updateDNSConfigMap(client controllerClient.Client, k8sclientSet clientset.I
...
-			expectedCorefile = expectedCorefile + lighthouseClusterIp + "\n" + "}\n"
-			superclusterCorefile := `supercluster.local:53 {
-    forward . `
-			expectedCorefile = expectedCorefile + superclusterCorefile + lighthouseClusterIp + "\n" + "}\n"
-			coreFile = expectedCorefile + coreFile
 		}
+		expectedCorefile := "#lighthouse-start AUTO-GENERATED SECTION. DO NOT EDIT\n"
+		for _, domain := range append([]string{"clusterset.local"}, cr.Spec.CustomDomains...) {
+			expectedCorefile = expectedCorefile + domain + `:53 {
+    forward . ` + lighthouseClusterIp + "\n" + "}\n"
```
Comment (tpantelis) [inline pkg/controller/servicediscovery/servicediscovery_controller.go:402]: Use `fmt.Sprintf` here as on L247?

**Review by aswinsuryan — DISMISSED** [review]
LGTM

File: pkg/controller/servicediscovery/servicediscovery_controller.go [hunk 2]
```diff
@@ -140,8 +142,8 @@ func (r *ReconcileServiceDiscovery) Reconcile(request reconcile.Request) (reconc
 		return reconcile.Result{}, err
 	}
 
-	ligthhouseConfigMap := newLigthhouseConfigMap(instance)
-	if _, err = helpers.ReconcileConfigMap(instance, ligthhouseConfigMap, reqLogger,
+	ligthhouseDnsConfigMap := newLigthhouseDnsConfigMap(instance)
```
Comment (skitt) [inline pkg/controller/servicediscovery/servicediscovery_controller.go:145]: This would be the ideal time to fix the “ligth” typo on both sides ;-).

**Review by skitt — CHANGES_REQUESTED** [review]

File: pkg/controller/servicediscovery/servicediscovery_controller.go (hunk 2 above)
Comment (vthapar) [reply pkg/controller/servicediscovery/servicediscovery_controller.go]: Done. There were more than just 2 :D

**Review by skitt — APPROVED** [review]

**Review by aswinsuryan — APPROVED** [review]

**Pull request merged**